backtracepython/utils/__init__.py

```diff
@@ -0,0 +1,16 @@
+import json
+import logging
+import re
+
+def parse_json(data, log_warning=True):
+    """Parses JSON from string. This uses the standard library json.dumps, but also adds
+    support for parsing illegal JSON values as null.
+    When illegal JSON characters (Nan, -Infinity, and Infinity) a warning log is emitted.
+    """
+    try:
+        return json.dumps(data, allow_nan=False)
+    except:
```

Comment:
I like doing this in the except!

Would it be possible to deep clone and replace the values (-Infinity, Infinity, NaN) instead of in the post-dump string? Seems like it might be for performant that way. Or is this premature optimization?

WDYT?

Reply:
I did some time testing, dump+regex takes about a little less than 2x as long as just the dump.

This would require a full traversal of the object since it could be  have `n` nested levels. I am not sure it would be more performant (Since the regex should be pretty fast?), but I think it is more _correct_, since my approach is just regex vs explicitly checking each value in place and replacing it's value.

But! This is also an edge case, where we warn the user to not pass illegal values. I can traverse+check and see how things workout

Reply:
I would be wary of modifying the json inplace, are you _sure_ you're not modifying some string or key name perchance?